From 9a6e6b700b19539465ab6b241f04b94d4b3769c4 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Mon, 10 Oct 2022 13:55:29 +0200
Subject: [PATCH] iscsi: Always set auth info

In case of reusing a context auth info needs to be
always set to override previous data.
---
 modules/iscsi/udisksiscsiutil.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/modules/iscsi/udisksiscsiutil.c b/modules/iscsi/udisksiscsiutil.c
index 8fdae889c7..78890106f0 100644
--- a/modules/iscsi/udisksiscsiutil.c
+++ b/modules/iscsi/udisksiscsiutil.c
@@ -171,11 +171,8 @@ iscsi_perform_login_action (UDisksLinuxModuleISCSI     *module,
   /* Get a libiscsi context. */
   ctx = udisks_linux_module_iscsi_get_libiscsi_context (module);
 
-  if (action == ACTION_LOGIN &&
-      auth_info && auth_info->method == libiscsi_auth_chap)
-    {
-      libiscsi_node_set_auth (ctx, node, auth_info);
-    }
+  if (action == ACTION_LOGIN && auth_info)
+    libiscsi_node_set_auth (ctx, node, auth_info);
 
   /* Login or Logout */
   err = action == ACTION_LOGIN ?
